### PR TITLE
getPeople implemented. Improvements to FanChart, AncestorWebs, FractalTree

### PR DESCRIPTION
**Pull request by Clarke-11007**
getPeople implemented for FanChart, AncestorWebs, FractalTree TREE APPS.
* FanChart - added ability to highlight by Category name (beta version of this feature, will be improved upon), also added option to display WikiTreeID or Ahnentafel # in cells.
* FractalTree - added Colour by Location options from FanChart (will add Category names option once perfected in FanChart)
* Ancestor Webs - added ability to add a second person to the web, and enabled the COMMON ancestors and SINGLES views to see the web of how two people are descended (often multiple ways) from their common ancestors.

### CODE REVIEW

File: views/fanChart/FanChartView.js [hunk 1]
```diff
@@ -1140,7 +1165,18 @@
...
+        highlightSelector.setAttribute("onchange", "FanChartView.optionElementJustChanged();");
+        let break4DNASelector = document.getElementById("highlight_options_break4DNA");
+        let howDNAlinksSelector = document.getElementById("highlight_options_howDNAlinks");
+        let catNameSelector = document.getElementById("highlight_options_catName");
+        let catNameSelectorLabel = document.getElementById("highlight_options_catName_label");
+        catNameSelector.style.display = "none";
+        catNameSelectorLabel.style.display = "none";
+        
+
+        console.log("TWEAKED the Highlights tab - how many categories I wonder ...", categoryList);
```
Comment (udjeni) [inline views/fanChart/FanChartView.js:1179]: There are lots of these (debugging) console logs throughout your code. Would it not be better to make use of the `condLog` available in tree.js? (Just remember to turn `debugLoggingOn` on in index.html, and off again before pushing a PR) Then people's browser memory will also be slightly less reduced by running your apps :)

**Review by udjeni — COMMENTED** [review]
I just browsed through the code and commented here and there on anything that caught my eye. I did not do a detailed review.

**Comment by Clarke-11007** [comment]
Thanks for those suggestions - yes - I did mean to remove those console.log
statements from there before committing - oops!  I don't think I've used
the .warn before, but will check that out too.

**Comment by udjeni** [comment]
Thanks. Looks good

File: views/webs/WebsView.js [hunk 2]
```diff
@@ -719,6 +719,16 @@
         return null;
     }
 
+    function acceptableDarkLineClr ( initClr ) {
+        if (initClr == "LightSkyBlue" || initClr == "HotPink") {
+            return initClr;    
+        }
+        let theClr = initClr;
+        if (ColourArray.indexOf(initClr) > -1) {
```
Comment (udjeni) [inline views/webs/WebsView.js:727]: No need to to do 2 `.indexOf` operations when just one would suffice. :)
`        const clrIdx = ColourArray.indexOf(initClr);
        if (clrIdx > -1) {
            theClr = LineColourArray[clrIdx];
        }
`

**Review by udjeni — APPROVED** [review]

**Comment by bcaseyrls** [comment]
Sorry, Greg. It looks like my merge of the previous pull request caused a conflict with your own getPeople addition to WikiTreeAPI.js. The differences look minimal, but I'll let you handle the resolution in case I'm missing something that would break your usage of the new action in your several views.

**Comment by Clarke-11007** [comment]
OK - thanks Brian - I'll check it out.

 - Greg

**Comment by Clarke-11007** [comment]
I believe that all conflicts have been resolved, and all of my Tree Apps that used the getPeople API now work with the revised version of it - plus FanDoku has been added to the list of TreeApps that use getPeople instead of getRelatives or getAncestors

**Pull request merged**